Rubberduck.Parsing/ParserRuleContextExtensions.cs

```diff
@@ -164,6 +164,111 @@ private static TContext GetAncestor_Recursive<TContext>(ParserRuleContext contex
             return listener.Matches;
         }
 
+        public static IEnumerable<T> GetChildren<T>(this RuleContext context)
+        {
+            if (context == null)
+            {
+                yield break;
+            }
+
+            for (var index = 0; index < context.ChildCount; index++)
+            {
+                var child = context.GetChild(index);
+                if (child is T)
+                {
+                    yield return (T)child;
+                }
+            }
+        }
+
+        public static bool HasParent(this RuleContext context, RuleContext parent)
+        {
+            if (context == null)
+            {
+                return false;
+            }
+            if (context == parent)
+            {
+                return true;
+            }
+            return HasParent(context.Parent, parent);
+        }
+
+        public static TContext FindChild<TContext>(this ParserRuleContext context) where TContext : ParserRuleContext
+        {
+            if (context == null)
+            {
+                return default;
+            }
+
+            for (var index = 0; index < context.ChildCount; index++)
+            {
+                var child = context.GetChild(index);
+                if (context.GetChild(index) is TContext)
+                {
+                    return (TContext)child;
+                }
+            }
+            return default;
+        }
+
+        public static bool HasChildToken(this IParseTree context, string token)
```

Comment:
You're mixing terms here. if you're checking the Text of a context against a "token" that has a special meaning for the parser. if you really only need to check against parser-level tokens, you should pass a Token (obtained from the VbaLexer) here. Otherwise this method's name should be adjusted to something like: HasChildWithText

Reply:
OK - agreed...that said, I am removing this method - unused in current PR.  It was a remnant of the retracted version.

Reply:
Removed method and modified callers to use `VBAParser` accessors (e.g., `context.TO()`) directly.